src/python/pants/backend/python/macros/common_fields.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Any, ClassVar, Dict, Iterable, Tuple
+
+from packaging.utils import canonicalize_name as canonicalize_project_name
+
+from pants.backend.python.target_types import normalize_module_mapping
+from pants.engine.addresses import Address
+from pants.engine.target import DictStringToStringSequenceField, OverridesField
+from pants.util.frozendict import FrozenDict
+
+
+class ModuleMappingField(DictStringToStringSequenceField):
+    alias = "module_mapping"
+    help = (
+        "A mapping of requirement names to a list of the modules they provide.\n\n"
+        'For example, `{"ansicolors": ["colors"]}`.\n\n'
+        "Any unspecified requirements will use the requirement name as the default module, "
+        'e.g. "Django" will default to `["django"]`.\n\n'
+        "This is used to infer dependencies."
```

Comment:
Should this explain itself in terms of the singular version of the field? https://www.pantsbuild.org/docs/reference-python_requirement#codemodulescode